kafka-connect-aws-s3/src/main/scala/io/lenses/streamreactor/connect/aws/s3/model/BucketPathPrefix.scala

```diff
@@ -41,10 +46,36 @@ case class BucketAndPrefix(
     .foreach(_ => throw new IllegalArgumentException("Nested prefix not currently supported"))
 }
 
-case class BucketAndPath(
+trait Location {
```

Comment:
Is there a reason for being empty and not have `def path:String`?
Can we express the stream and local "location" in a similar way? It looks like an if stream then `nothing really here` else if local folder path

Reply:
There are 3 different types of location and we don't necessarily use `path` in each one.

But I've added it to some extent.